subscription_web_access/models/res_users.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2019 Coop IT Easy SCRL fs
+#   Robin Keunen <robin@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+import json
+
+from odoo import api, models
+from odoo.exceptions import MissingError
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    @api.model
+    def get_subscription(self, user_id):
+        user = self.env["res.users"].browse(user_id)
+        if not user:
+            return json.dumps({"error": "user not found"})
+
+        try:
+            partner = user.partner_id
+        except MissingError:
+            return json.dumps(
+                {
+                    "id": user_id,
+                    "start": "",
+                    "end": "",
+                    "subscription": "",
+                    "subscribed": False,
+                }
+            )
+
+        for contract in partner.contract_ids:
+            for line in contract.contract_line_ids:
+                if line.state == "in-progress" or line.state == "upcoming-close":
```

Comment:
this is exactly the same code as used in `res.partner._compute_subscription_status()`. this could be refactored either as a computed field (something like `ongoing_contract_line_id`) or as a method (something like `.get_ongoing_contract_line()`). both should return an empty `contract.line` recordset in case there is no ongoing contract line. i think the computed field option is better.

Reply:
OK I add a comment in the code for later improvement